server/game/cards/07-GR/Anachronometer.js

```diff
@@ -0,0 +1,47 @@
+const Card = require('../../Card.js');
+
+function anachronometerAbility(ability) {
+    return {
+        effect:
+            'shuffle their discard into their deck, draw a card, and discard the top {1} cards of the deck',
+        effectArgs: (context) => [
+            context.player.opponent ? context.player.opponent.discard.length : 0
+        ],
+        gameAction: ability.actions.sequential([
+            ability.actions.returnToDeck((context) => ({
+                shuffle: true,
+                target: context.player.discard
+            })),
+            ability.actions.draw(),
+            ability.actions.conditional({
+                condition: (context) => !!context.player.opponent,
+                trueGameAction: ability.actions.discard((context) => ({
+                    target: context.player.deck.slice(
+                        0,
+                        Math.min(context.player.opponent.discard.length, context.player.deck.length)
+                    )
+                }))
+            })
+        ])
+    };
+}
+
+class Anachronometer extends Card {
+    // This creature gains: "After Fight/After Reap: Shuffle your
```

Comment:
Interesting, is this the first we have a double gain ability when using upgrades?

Reply:
You mean for both fight and reap?  That's somewhat common, like https://archonarcana.com/Observ-u-Max .